docs/specs/clients/sign/client-api.md

```diff
@@ -88,5 +132,10 @@ abstract class Client {
 
   // subscribe to session delete
   public abstract on("session_delete", (sessionDelete: { id: number, topic: string }) => {}): void;
+
+  // ---------- Wallet Authentication Events ----------------------------------------------- //
+
+  // wallets to subscribe to session authenticated
+  public abstract on("session_authenticate", (sessionAuthenticateRequest: SessionAuthenticateRequest) => {}): void;
```

Comment:
this is SessionAuthenticateRequest and you don't have on response event, does mean that if app crashes you are loosing track of the response?

Reply:
Shouldn't that be cached internally with the SignClient? You can use the `getPnedingAuthRequests`

Reply:
We should have "session_authenticate_response" for a dapp